### ISSUE TITLE
Migration init job hangs when using your own DB

### ISSUE THREAD

**Issue by yasn77** [giantswarm/kong-app#38]
If using your own DB and set `postgresql.enabled: false` then  the environment variables`KONG_PG_HOST` and/or `KONG_PG_PORT` might not be set. So the migration init script probably will hang, as it runs `nc` command with these environment variables.

https://github.com/giantswarm/kong-app/blob/master/helm/kong-app/templates/_helpers.tpl#L435 Is how environment variables are populated

https://github.com/giantswarm/kong-app/blob/master/helm/kong-app/templates/_helpers.tpl#L493 `nc` command that runs inside of init migrations job

**Comment by rathonit01** [comment]
Hi @yasn77 ,
today we upgraded helm chart for installation of custom plugin via configmap 

```
#values.yaml content 
plugins:
  configMaps:
    - name: key-auth-secret-root
      pluginName: key-auth-secret
    - name: key-auth-secret-migrations
      pluginName: key-auth-secret
      path: key-auth-secret/migrations
```


help upgrade got failed with error (mesg below)

```
#snapshot of error message
locker:(*core.FlockerVolumeSource)(nil), DownwardAPI:(*core.DownwardAPIVolumeSource)(nil), FC:(*core.FCVolumeSource)(nil), AzureFile:(*core.AzureFileVolumeSource)(nil), ConfigMap:(*core.ConfigMapVolumeSource)(nil), VsphereVolume:(*core.VsphereVirtualDiskVolumeSource)(nil), AzureDisk:(*core.AzureDiskVolumeSource)(nil), PhotonPersistentDisk:(*core.PhotonPersistentDiskVolumeSource)(nil), Projected:(*core.ProjectedVolumeSource)(nil), PortworxVolume:(*core.PortworxVolumeSource)(nil), ScaleIO:(*core.ScaleIOVolumeSource)(nil), StorageOS:(*core.StorageOSVolumeSource)(nil), CSI:(*core.CSIVolumeSource)(nil)}}, core.Volume{Name:"kong-plugin-key-auth-secret", VolumeSource:core.VolumeSource{HostPath:(
```

this is happening because init migration jobs was already present and its not allowing to run config map changes.

solution: need to delete existing migration jobs from namespace where kong is deployed. and upgrade with helm.

#####Next step:
is it possible to keep deletion of jobs via helm chart during upgrade process. 
we are thinking to make it part of Helm chart so that as soon as we run helm it first checks whether jobs are available or not, if available delete existing init jobs.

**Comment by yasn77** [comment]
@rathonit01 

Interesting, because on our call I was expecting this to come up but it didn't... I don't see the full error message, try using [triple backticks](https://help.github.com/en/github/writing-on-github/creating-and-highlighting-code-blocks#fenced-code-blocks)

But I am suspecting it is due to immutable fields in the job. A workaround for now is `helm upgrade --set runMigrations=false ...`, I think I have a fix, will try get an update to you tomorrow.

Thanks

**Comment by yasn77** [comment]
This issue is fixed in #39

**Comment by yasn77** [comment]
@rathonit01 

*I think* your issue has been resolved in #40 

If it hasn't please open a new issue, as I don't think it is related.

Thanks
